packages/ckb-indexer/src/collector.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  Script,
+  utils,
+  Cell,
+  BaseCellCollector,
+  ScriptWrapper,
+  Hexadecimal,
+} from "@ckb-lumos/base";
+import { validators } from "ckb-js-toolkit";
+import {
+  AdditionalOptions,
+  CkbQueryOptions,
+  GetCellsResults,
+  HexadecimalRange,
+  Order,
+  SearchFilter,
+} from "./indexer";
+
+// import { logger } from "./logger";
+import { CkbIndexer, ScriptType, SearchKey } from "./indexer";
+
+export class IndexerCollector implements BaseCellCollector {
+  lock!: Script;
+  type!: Script | string;
+  data: string | null;
+  fromBlock: string | null;
+  toBlock: string | null;
+  order: Order;
+  skip: number | null;
+  argsLen: number | "any";
+  outputDataLenRange: HexadecimalRange | null;
+  outputCapacityRange: HexadecimalRange | null;
+  sizeLimit: number | undefined;
+
+  constructor(public indexer: CkbIndexer, public queries: CkbQueryOptions) {
+    const {
+      lock = null,
+      type = null,
+      argsLen = -1,
+      data = "any",
+      fromBlock = null,
+      toBlock = null,
+      order = Order.asc,
+      skip = null,
+      outputDataLenRange = null,
+      outputCapacityRange = null,
+      sizeLimit = undefined,
+    } = queries;
+    if (!lock && (!type || type === "empty")) {
+      throw new Error("Either lock or type script must be provided!");
+    }
+
+    function instanceOfScriptWrapper(object: any): object is ScriptWrapper {
```

Comment:
```suggestion
    function instanceOfScriptWrapper(object: unknown): object is ScriptWrapper {
```

Use `unknown` instead of `any` whenever possible